tests/test_scope.js

```diff
@@ -86,3 +87,82 @@ test('loadDefs throws expected when fs is not available', t => {
 
   t.end()
 })
+
+test('filter path with function', async t => {
+  const scope = nock('http://example.test')
+    .filteringPath(() => '/?a=2&b=1')
+    .get('/?a=2&b=1')
+    .reply(200, 'Hello World!')
+
+  const { statusCode } = await got('http://example.test/', {
+    query: { a: '1', b: '2' },
+  })
```

Comment:
These are fine for now (and I know you didn't write them, so this is just a thought for the future!

It's hard to tell from these tests what exactly the feature is for. They don't have descriptive titles, and as far as I can tell they are all positive tests – there's nothing that checks what happens when the request doesn't match the filter.

Reply:
I had the same thought, but didn't want to blow up the scope of this PR.